test/groovy/WriteTemporaryCredentialsTest.groovy

```diff
@@ -0,0 +1,128 @@
+        .around(stepRule)
+        .around(writeFileRule)
+        .around(fileExistsRule)
+        .around(readFileRule)
+        .around(shellRule)
+
+    @Before
+    void init() {
+        bodyExecuted = false
+
+        helper.registerAllowedMethod("deleteDir", [], null)
+
+        credentialsRule.reset()
+            .withCredentials('erp-credentials', 'test_user', '********')
+            .withCredentials('testCred2', 'test_other', '**')
+    }
+
+    @Test
+    void noCredentials() {
+        nullScript.commonPipelineEnvironment.configuration = [stages: [myStage:[
+            credentialsDirectory: './',
+        ]]]
+        stepRule.step.writeTemporaryCredentials(
+            script: nullScript,
+            stageName: "myStage",
+        ){
+            bodyExecuted = true
+        }
+        assertTrue(bodyExecuted)
+        assertThat(writeFileRule.files.keySet(), hasSize(0))
+        assertThat(shellRule.shell, hasSize(0))
+    }
+
+    @Test
+    void credentialsNoList() {
+        def credential = "id"
+
+        nullScript.commonPipelineEnvironment.configuration = [stages: [myStage:[
+            credentials: credential
+        ]]]
+
+        thrown.expect(hudson.AbortException)
+        thrown.expectMessage('[writeTemporaryCredentials] The execution failed, since credentials is not a list. Please provide credentials as a list of maps.')
+
+        stepRule.step.writeTemporaryCredentials(
+            script: nullScript,
+            stageName: "myStage",
+        ){
+            bodyExecuted = true
+        }
+    }
+
+    @Test
+    void noCredentialsDirectory() {
+        def credential = [alias: 'ERP', credentialId: 'erp-credentials']
+
+        nullScript.commonPipelineEnvironment.configuration = [stages: [myStage:[
+            credentials: [credential]
+        ]]]
+
+        thrown.expect(hudson.AbortException)
+        thrown.expectMessage("[writeTemporaryCredentials] The execution failed, since no credentialsDirectory is defined. Please provide the path for the credentials file.")
+
+        stepRule.step.writeTemporaryCredentials(
+            script: nullScript,
+            stageName: "myStage",
+        ){
+            bodyExecuted = true
+        }
+    }
+
+    @Test
+    void credentialsFileWrittenAndRemoved() {
+        def credential = [alias: 'ERP', credentialId: 'erp-credentials']
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        nullScript.commonPipelineEnvironment.configuration = [stages: [myStage:[
+            credentials: [credential],
+            credentialsDirectory: './',
+        ]]]
+
+        stepRule.step.writeTemporaryCredentials(
+            script: nullScript,
+            stageName: "myStage",
+        ){
+            bodyExecuted = true
+        }
+
+        assertThat(writeFileRule.files['credentials.json'], containsString('"alias":"ERP","username":"test_user","password":"********"'))
+        assertThat(shellRule.shell, hasItem('rm -f credentials.json'))
```

Comment:
```suggestion
        assertThat(shellRule.shell, hasItem('rm -f credentials.json'))
        assertTrue(bodyExecuted)
```

Reply:
Makes sense. Will be addressed with the next commit.